cache/cache.cpp

```diff
@@ -555,25 +559,22 @@ static int dnet_cmd_cache_io_lookup_new(struct cache_manager *cache,
 		}
 	}
 
-	auto response = serialize(dnet_lookup_response{
-		0, // record_flags
-		it.user_flags, // user_flags
-		"", // path
-
-		it.json_timestamp, // json_timestamp
-		0, // json_offset
-		it.json->size(), // json_size
-		it.json->size(), // json_capacity
-		std::move(json_checksum), // json_checksum
-
-		it.timestamp, // data_timestamp
-		0, // data_offset
-		it.data->size(), // data_size
-		std::move(data_checksum), // data_checksum
-	});
-
-	cmd->flags &= ~DNET_FLAGS_NEED_ACK;
-	return dnet_send_reply(st, cmd, response.data(), response.size(), 0, context);
+	std::unique_ptr<n2::lookup_response>
+		response(new n2::lookup_response(request->cmd,
```

Comment:
Let's at least call `new` with `std::nothrow` and check whether it has returned `nullptr`.

Reply:
Fixed